server/routes/user.js

```diff
@@ -1,77 +1,28 @@
 const express = require('express');
 const router = express.Router();
+const { ApiService } = require('../services');
 const User = require('../database/models/user');
-const passport = require('../passport');
 
-router.post('/', (req, res) => {
-    console.log('user signup');
+router.get('/', async (req, res) => {
+  // validate auth token (return 401 if invalid)
+  const headers = { Authorization: `${req.get('Authorization')}` };
+  const url = `${process.env.AWS_COGNITO_BASE_URL}/oauth2/userInfo`;
+  const userInfoResponse = await ApiService.get(url, { headers }).catch(() => res.sendStatus(401));
 
-    const { username, password, name, email } = req.body;
-    // ADD VALIDATION
-    User.findOne({ username: username }, (err, user) => {
-        if (err) {
-            console.log('User.js post error: ', err);
-        } else if (user) {
-            res.json({
-                error: `Sorry, already a user with the username: ${username}`,
-            });
-        } else {
-            const newUser = new User({
-                username: username,
-                password: password,
-                name: name,
-                email: email,
-            });
-            newUser.save((err, savedUser) => {
-                if (err) return res.json(err);
-                res.json(savedUser);
-            });
-        }
-    });
-});
-
-router.get('/data/:username', (req, res, next) => {
-    // console.log("this is for user: " + req.params.username)
-    User.findOne({ username: req.params.username }).then(
-        (userData) => res.json(userData),
-        (error) => res.sendStatus(404)
-    );
-});
-
-router.post(
-    '/signin',
-    function (req, res, next) {
-        console.log('routes/user.js, login, req.body: ');
-        console.log(req.body);
-        next();
-    },
-    passport.authenticate('local'),
-    (req, res) => {
-        console.log('logged in', req.user);
-        var userInfo = {
-            username: req.user.username,
-        };
-        res.send(userInfo);
-    }
-);
-
-router.get('/', (req, res, next) => {
-    console.log('===== user!!=====');
-    console.log(req.user);
-    if (req.user) {
-        res.json({ user: req.user });
-    } else {
-        res.json({ user: null });
-    }
-});
+  // get username & email from valid token response
+  const { username, email } = userInfoResponse.data;
 
-router.post('/logout', (req, res) => {
-    if (req.user) {
-        req.logout();
-        res.send({ msg: 'logging out' });
-    } else {
-        res.send({ msg: 'no user to log out' });
+  // return user data from DB (create if necessary, return 500 on error)
+  try {
+    let user = await User.findOne({ username });
+    if (!user) {
+      user = await new User({ username, email }).save();
     }
+    res.json(user);
+  } catch (error) {
+    console.log(error);
```

Comment:
@LGG233 This server logging is intentional. We can "tail" the Heroku server logs for errors like this.